### PR TITLE
Tensor construction codemod(ResizeLike) - 1/7

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
Codemod generated with clangr shard mode, 25 files per diff,
motivation: https://github.com/pytorch/pytorch/pull/12407

Differential Revision: D13419563

### CODE REVIEW

**Comment by ezyang** [comment]
In these diffs can we *please* have a description which explains (1) what the codemod applied is, (2) why we want to apply this codemod and (3) why it is semantics preserving.

File: caffe2/core/operator.h [hunk 1]
```diff
@@ -613,6 +616,18 @@ class Operator : public OperatorBase {
     return OperatorBase::template Output<Tensor>(idx, type);
   }
 
+  Tensor* OutputTensorCopyFrom(
```
Comment (ezyang) [inline caffe2/core/operator.h]: This looks dead. Did you mean to put this in a different diff?

**Comment by ezyang** [comment]
> In these diffs can we please have a description which explains (1) what the codemod applied is, (2) why we want to apply this codemod and (3) why it is semantics preserving.

Actually, the codemod here was fine; the problem was the changes in caffe2/core/operator.h which were not explained.

File: caffe2/core/operator.h (hunk 1 above)
Comment (jerryzh168) [reply caffe2/core/operator.h]: OK, will put in a different diff

**Comment by jerryzh168** [comment]
@ezyang (1) and (2) are in summary((1) is only visible to Facebook)

File: caffe2/core/operator.h [hunk 2]
```diff
@@ -208,7 +208,10 @@ class CAFFE2_API OperatorBase : public Observable<OperatorBase> {
       at::TensorOptions options,
       const Tensor& src,
       bool async = false) {
-    Tensor* t = Output<Tensor>(idx, options.device().type());
+    if (!options.has_dtype()) {
+      options = options.dtype(src.dtype());
+    }
+    Tensor* t = OutputTensor(idx, src.sizes(), options);
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply caffe2/core/operator.h]: probably this is for another diff.. I'll split

**Pull request closed without merging**